lib/ExecutionEngine/ExecutionEngine2.cpp

```diff
@@ -109,6 +109,7 @@ void ExecutionEngine2::runInternal(ExecutionContext &context,
   std::promise<void> runPromise;
   auto fut = runPromise.get_future();
   llvm::Error runErr = llvm::Error::success();
+  (void)(bool) runErr;
```

Comment:
I'm kind of unhappy with this; it feels like the host/device manager API fights with how `llvm::Error` is designed to work.  When  we create `runErr` it's really just not initialized, but we have to pointlessly "check" it so it can be overwritten without errors later...

Reply:
could we use this macro? MARK_ERR_CHECKED

Reply:
Done!